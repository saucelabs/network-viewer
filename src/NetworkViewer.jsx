import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames/bind';

import NetworkProvider from './state/network/NetworkProvider';
import MainContainer from './Containers/MainContainer';
import Styles from './NetworkViewer.styles.scss';
import ThemeProvider from './state/theme/Context';

const contextClassNames = classNames.bind(Styles);

const NetworkViewer = ({
  file,
  data,
  fetchOptions,
  scrollTimeStamp,
  options,
  onPause,
  onRequestSelect,
  onResume,
  onReset,
  scrollRequestPosition,
  autoHighlightChange,
  onDataLoaded,
  onDataError,
  containerClassName,
}) => (
  <section className={contextClassNames('network-viewer', containerClassName)}>
    <ThemeProvider options={options}>
      <NetworkProvider
        autoHighlightChange={autoHighlightChange}
        data={data}
        fetchOptions={fetchOptions}
        file={file}
        onDataError={onDataError}
        onDataLoaded={onDataLoaded}
        onPause={onPause}
        onReset={onReset}
        onResume={onResume}
        scrollRequestPosition={scrollRequestPosition}
        scrollTimeStamp={scrollTimeStamp}
      >
        <MainContainer onRequestSelect={onRequestSelect} />
      </NetworkProvider>
    </ThemeProvider>
  </section>
);

NetworkViewer.propTypes = {
  autoHighlightChange: PropTypes.bool,
  containerClassName: PropTypes.string,
  data: PropTypes.object,
  fetchOptions: PropTypes.object,
  file: PropTypes.string,
  onDataError: PropTypes.func,
  onDataLoaded: PropTypes.func,
  onPause: PropTypes.func,
  onRequestSelect: PropTypes.func,
  onReset: PropTypes.func,
  onResume: PropTypes.func,
  options: PropTypes.object,
  scrollRequestPosition: PropTypes.oneOf(['before', 'after', 'near']),
  scrollTimeStamp: PropTypes.number,
};

NetworkViewer.defaultProps = {
  autoHighlightChange: false,
  containerClassName: null,
  data: null,
  fetchOptions: { withCredentials: true },
  file: null,
  onDataError: null,
  onDataLoaded: null,
  onPause: null,
  onRequestSelect: () => {},
  onReset: null,
  onResume: null,
  options: null,
  scrollRequestPosition: 'near',
  scrollTimeStamp: null,
};

export default NetworkViewer;
