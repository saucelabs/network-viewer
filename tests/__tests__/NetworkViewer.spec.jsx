import React from 'react';
import { shallow } from 'enzyme';

import NetworkViewer from './../../src/NetworkViewer';

it('renders without crashing', () => {
  const element = shallow(<NetworkViewer />);
  expect(element).toMatchSnapshot();
});
