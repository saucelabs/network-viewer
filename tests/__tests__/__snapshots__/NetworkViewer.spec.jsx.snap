// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<section
  className="network-viewer"
>
  <ThemeProvider
    options={null}
  >
    <NetworkProvider
      autoHighlightChange={false}
      data={null}
      fetchOptions={
        Object {
          "withCredentials": true,
        }
      }
      file={null}
      initialState={
        Immutable.Map {
          "search": Object {
            "name": "URL",
            "value": "",
          },
          "error": null,
          "statusFilter": Object {
            "name": "-",
            "value": null,
          },
          "scrollToIndex": null,
          "data": Immutable.List [],
          "selectedReqIndex": null,
          "reqDetail": null,
          "showReqDetail": false,
          "typeFilter": Object {
            "name": null,
            "value": null,
          },
          "rawData": null,
          "totalNetworkTime": null,
          "actualData": Immutable.List [],
          "dataSummary": Immutable.Map {},
          "loading": false,
          "sort": Object {
            "isAcs": true,
            "key": "startedDateTime",
          },
        }
      }
      onDataError={null}
      onDataLoaded={null}
      onPause={null}
      onReset={null}
      onResume={null}
      scrollRequestPosition="near"
      scrollTimeStamp={null}
    >
      <MainContainer
        onRequestSelect={[Function]}
      />
    </NetworkProvider>
  </ThemeProvider>
</section>
`;
